operation/fhir-operation-versions/src/main/java/com/ibm/fhir/operation/versions/VersionsOperation.java

```diff
@@ -0,0 +1,189 @@
+        // Set properties in the operation context that are used to return this response instead of Parameters
+        operationContext.setProperty(FHIROperationContext.PROPNAME_RESPONSE, response);
+        operationContext.setProperty(FHIROperationContext.PROPNAME_STATUS_TYPE, STATUS_OK);
+        return FHIROperationUtil.getOutputParameters(null);
+    }
+
+    /**
+     * Builds a response when output is XML (non-FHIR).
+     * @param operationContext the operation context
+     * @param versions the supported FHIR versions
+     * @param defaultVersion the default FHIR version
+     * @return a parameters object that will not be the response entity
+     * @throws ParserConfigurationException
+     */
+    private Parameters buildXmlResponse(FHIROperationContext operationContext, List<String> versions, String defaultVersion) throws ParserConfigurationException {
+        // Build the response
+        DocumentBuilder docBuilder = XML_DOM_FACTORY.newDocumentBuilder();
+        Document doc = docBuilder.newDocument();
+        Element versionsElement = doc.createElement(PARAM_VERSIONS);
+        doc.appendChild(versionsElement);
+        for (String version : versions) {
+            Element versionElement = doc.createElement(PARAM_VERSION);
+            versionElement.setTextContent(version);
+            versionsElement.appendChild(versionElement);
+        }
+        Element defaultElement = doc.createElement(PARAM_DEFAULT);
+        defaultElement.setTextContent(defaultVersion);
+        versionsElement.appendChild(defaultElement);
+        Response response = Response.status(STATUS_OK).entity(doc).build();
+        // Set properties in the operation context that are used to return this response instead of Parameters
+        operationContext.setProperty(FHIROperationContext.PROPNAME_RESPONSE, response);
+        operationContext.setProperty(FHIROperationContext.PROPNAME_STATUS_TYPE, STATUS_OK);
+        return FHIROperationUtil.getOutputParameters(null);
+    }
+
+    /**
+     * Gets the non-FHIR format for returning output. If the first Accept header
+     * is either 'application/json' or 'application/xml', the output is returned in
+     * that non-FHIR format.
+     * @param operationContext the operation context
+     * @return 'application/json', 'application/xml', or null
+     */
+    private String getNonFhirOutputFormat(FHIROperationContext operationContext) {
+        String headerString = operationContext.getHeaderString(ACCEPT_HEADER);
+        if (headerString != null) {
+            for (String headerStringElement : headerString.split(",")) {
+                if (MediaType.APPLICATION_JSON.equals(headerStringElement)) {
+                    return MediaType.APPLICATION_JSON;
+                }
+                if (MediaType.APPLICATION_XML.equals(headerStringElement)) {
+                    return MediaType.APPLICATION_XML;
+                }
```

Comment:
To be robust to content-type and other MIME-type parameters, we should probably split the headerStringElement strings by ";", take just the first component, then trim().